### ISSUE TITLE
synctex path finding broken on windows

### ISSUE THREAD

**Issue by sunderme** [jlaurens/synctex#63]
Concerns current git head.
```
last_component = PathFindFileName(string);
last_extension = PathFindExtension(string);
```
Don't work as they assume UTF16 string and fail with 8bit coding.
Tested with msys64 (ucrt) and vcpkg.

**Comment by sunderme** [comment]
(needs to be PathFindFileNameA(string);...)

**Comment by jlaurens** [comment]
We know for sure that XeTeX and LuaTeX use unicode, but for the other engines, things are not clear to my mind. Moreover, things might be very different for pdftex and ptex at least. Digging the sources is not an easy task, I should ask tex.stackexchange.

**Comment by ratijas** [comment]
> We know for sure that XeTeX and LuaTeX use unicode

when you say "unicode" it's a good idea to specify which flavor of encoding is on your mind.  I assume you meant UTF-8.

IMHO it would be a good idea to force everyone to use UTF-8, and forget code pages like a bad deam they are.

**Comment by jlaurens** [comment]
XeTeX and LuaTeX basically use 16 bits to store characters internally whereas the other engines only use 8 bits. At the same time these "unicode" aware engines use UTF8 for input and file paths...

Forcing everyone to use UTF8 is definitely an option, but this must be done for each different TeX engine (except TeX itself because it is frozen) which is a rather long task.

**Comment by ThiloteE** [comment]
So doing this for every TeX engine will be done by whom?

**Comment by ThiloteE** [comment]
Forcing UTF8 and making code pages optional for people that require backwards compatibility sounds reasonable to me, but I am biased for sure.

**Comment by jlaurens** [comment]
The opinion of MikTeX maintainer on that matter would be highly valuable

**Comment by koppor** [comment]
> At the same time these "unicode" aware engines use UTF8 for input and file paths...

True: As of April 2018: The community has changed its default encoding to UTF-8 as of April 2018 (see https://tug.org/TUGboat/tb39-1/tb121ltnews28.pdf).

Source: https://tex.stackexchange.com/a/370279/9075
